### ISSUE TITLE
error_detection_glo is not used (no inversion of incorrect bit)

### ISSUE THREAD

**Issue by ValZapod** [swift-nav/libswiftnav#15]
It is only used in a unitest and other wise ret value is not stored.

**Comment by ValZapod** [comment]
@RReichert

**Comment by ValZapod** [comment]
@silverjam ping!

**Comment by silverjam** [comment]
@ValZapod forgive my ignorance, the code avoids decoding if the function returns false, where does the information need to be stored?

https://github.com/swift-nav/libswiftnav/blob/aabca50f70c1c9dca8b6afb0f70d33f9e62ba037/src/decode_glo.c#L261

**Comment by silverjam** [comment]
Also, to clarify intent, libswiftnav is essentially a collection of functions and data types intended for use in a larger GNSS application, the fact that a function isn't necessarily used within the code base (except for a unit test) isn't necessarily an error.

**Comment by ValZapod** [comment]
> code avoids decoding if the function returns false

Yes, but that is not all it should do. It actually can returns a bit that should be inverted (0 to 1, 1 to 0). That is what is called self correction, whyle you do implement that return of that bit, you are treating that case as bad as well, while you should just invert the bit and treat it as good.

> isn't necessarily used within the code base (except for a unit test) isn't necessarily an error.

It is used in code, but incorrectly.

**Comment by jbangelo** [comment]
Hello @ValZapod , sorry about the misunderstanding initially. We see the problem you are referring to, and agree that we could improve the GLONASS ephemeris decoding by fixing single bit errors when they are detected. I guess we've not see enough single bit errors in our Piksi Multi product to notice this inefficiency. Thanks for pointing it out to us!

Unfortunately we're a little busy at the moment and probably won't be able to get to addressing this in the near term. If this is an immediate concern for you, and you have a patch that improves this behavior, feel free to submit it as a PR for us to review (we usually ask for at least one new test to exercise the new behavior before accepting and merging PRs).

**Comment by ValZapod** [comment]
> usually ask for at least one new test to exercise the new behavior

LOL, that is point, the unittest does actually test for this ALREADY.

https://github.com/swift-nav/libswiftnav/blob/aabca50f70c1c9dca8b6afb0f70d33f9e62ba037/tests/check_decode_glo.c#L189-L192

As for the patch you will have to de'const this and modify it on the fly:

https://github.com/swift-nav/libswiftnav/blob/aabca50f70c1c9dca8b6afb0f70d33f9e62ba037/src/decode_glo.c#L254

**Comment by ValZapod** [comment]
We implemented it in gnss-sdr. Any updates?
